src/dataset.rs

```diff
@@ -14,10 +16,10 @@ pub struct Dataset {
     last_country: Option<String>,
     last_region: Option<String>,
     last_site: Option<String>,
-    countries: Vec<String>,
-    regions: Vec<String>,
-    sites: Vec<String>,
-    devices: Vec<String>,
+    countries: HashSet<String>,
```

Comment:
@ntlhui Was there a purpose for storing the locations at the dataset level? I can't seem to find where this is used in the Python codebase.

Reply:
His response: currently only used to save location information at the Dataset level. We may want to add Dataset level metadata as well but we can do that in a future PR.